### PR TITLE
feat: extend bidding engine

### PR DESCRIPTION
**Pull request by palozano**
chore: create pull request

# Summary of changes

- Extends bidding engine's features. 
- Adds tracking between side effects and bidders.
-  Adds scenarios for the executor to behave differently based on some configuration. 
- Adds configuration loading. 
- Adds cleaning for the Maps tracking relationships between sfx and bidders. 
- Adds tests for the new features, and tests for previous code.

## Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue):
    - [ ] Have you **written tests **to protect against future occurrences of the bug?
- [x] New feature (non-breaking change which adds functionality)
    - [ ] Have you **seen it working** in a development environment?
    - [X] Have you **written tests** for **happy** and **unhappy** paths?
    - [ ] Have you implemented this feature as per the **issue acceptance criteria**?
- [ ] Breaking change (a feature that would cause existing functionality not to work as expected
    - [ ] Is the breaking change **documented**?
    - [ ] Has any previous changes been **deprecated**?
- [ ] CI/CD
    - [ ] If introducing new actions, have you **looked at the action code** for anything spurious?
    - [ ] Have you written **custom scripts** for things that could be actions already on the marketplace?
    - [ ] If introducing new actions, have you **pegged a static version**?
- [ ] Documentation Update
    - [ ] Have you checked other documentation, such as the docs repository?
    - [ ] If updating script documentation, have you **tested it on both environments**?
- [ ] Substrate
    - [ ] RPC methods?
    - [ ] Chainspec, both JSON specs & the module?
    - [ ] Runtime versioning?
    - [ ] Benchmarks?
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Tests can, and should, be extended.

### CODE REVIEW

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:338]: I would add this to processBid I think, then we ensure one source of these, the incoming events

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:416]: xtxId is the name for the id of the XTX object, not the executor (each Execution class instance has an xtxId which is generated by the circuit). Please rename accordingly!

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:419]:
```suggestion
            if (previousXtxBids) {
```

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:440]:
```suggestion
```

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:338]:
```suggestion
```

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:444]: rename also

**Review by petscheit — CHANGES_REQUESTED** [review]
Looks sick! A couple of minor things, LGTM after!

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (palozano) [reply client/packages/executor/src/executionManager/sideEffect.ts:338]: Done! ✌️

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (palozano) [reply client/packages/executor/src/executionManager/sideEffect.ts:444]: Done! ✌️
I removed the method there because it's already present in the bidding engine.

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (palozano) [reply client/packages/executor/src/executionManager/sideEffect.ts:416]: Done! ✌️
I messed up names, I don't know why. Removed the method there because it's already present in the bidding engine.

**Review by 3h4x — CHANGES_REQUESTED** [review]
I'm blocking this PR so noone preemptively merge it because test should have GHA.
Working with Pablo on it.

**Review by 3h4x — DISMISSED** [review]
GHA running tests added.

File: .github/workflows/test-executor.yml
Comment (petscheit) [inline .github/workflows/test-executor.yml:22]: CI is failing because you need to build local deps and install. You can use the makefile for the local ones (in packages) and after run a yarn

Comment (3h4x) [reply .github/workflows/test-executor.yml:22]: Correcto!

File: client/packages/executor/config/config.ts
Comment (petscheit) [reply client/packages/executor/config/config.ts:154]: Please use the ´BiddingStrategy´ type here you created

File: client/packages/executor/src/executionManager/execution.ts
Comment (petscheit) [inline client/packages/executor/src/executionManager/execution.ts:169]: This shouldnt work tbh. The status is called `readyToExecute` (https://github.com/t3rn/t3rn/blob/development/client/packages/sdk/src/side-effects/types.ts)

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:74]: Same here, InBidding is the correct type

File: client/packages/executor/src/executionManager/sideEffect.ts
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:322]: Here too

File: client/packages/sdk/src/side-effects/types.ts
Comment (petscheit) [inline client/packages/sdk/src/side-effects/types.ts:16]: Please revert

**Review by petscheit — DISMISSED** [review]

**Review by MaciejBaj — DISMISSED** [review]

**Review by petscheit — APPROVED** [review]

**Review by chiefbiiko — APPROVED** [review]

**Review by MaciejBaj — APPROVED** [review]

**Review by chiefbiiko — APPROVED** [review]
LGTM

**Review by petscheit — APPROVED** [review]

**Pull request merged**
